src/Refactoring-Core/RBIsValidSharedVariableName.class.st

```diff
@@ -0,0 +1,21 @@
+"
+I check if a name is valid for an instance variable
+"
+Class {
+	#name : 'RBIsValidSharedVariableName',
+	#superclass : 'RBVariableNameCondition',
+	#category : 'Refactoring-Core-Conditions',
+	#package : 'Refactoring-Core',
+	#tag : 'Conditions'
+}
+
+{ #category : 'checking' }
+RBIsValidSharedVariableName >> check [
+	| string |
+	name isString ifFalse: [^false].
+	string := name asString.
```

Comment:
I would test first the not uppercase.
Now I wonder why such functionality is not just defined in the superclass RBAbstractTransformation because it does not check that there is already an instance variable in the hierarchy for example.
It is just a textual check independent of any model.

Reply:
As we spoke in the last meeting, we are keeping validations inside the conditions. But yes, syntactic validations should be different than validations depending on the model or other objects